libp2p/protocols/pubsub/pubsub.nim

```diff
@@ -14,13 +14,20 @@ import pubsubpeer,
        ../protocol,
        ../../connection,
        ../../peerinfo
+import metrics
 
 export PubSubPeer
 export PubSubObserver
 
 logScope:
   topic = "PubSub"
 
+declareGauge(libp2p_pubsub_peers, "pubsub peer instances")
+declareGauge(libp2p_pubsub_topics, "pubsub subscribed topics")
+declareGauge(libp2p_pubsub_validation_success, "pubsub successfully validated messages")
```

Comment:
these should be counters: https://prometheus.io/docs/concepts/metric_types/

Reply:
Indeed! I was gonna investigate cos I didn't like `inc` for some of the types. Thanks.

Reply:
@arnetheduck are you sure tho?
Read that page says explicitly: 
> Do not use a counter to expose a value that can decrease

Reply:
I agree that validation should be counter tho, but peers and topics no